### ISSUE TITLE
find_all_snippet_files and _file_to_edit search different paths

### ISSUE THREAD

**Issue by froody** [SirVer/ultisnips#907]
TL;DR: `_file_to_edit looks` in `$PWD` by way of `os.path.realpath()`, `find_all_snippet_files` does not look in `$PWD`, so `:UltiSnipsEdit` opens a file that is not read by `UltiSnips_Manager.expand()`

On a default install, with no setting of UltiSnips vars in my .vimrc other than:
```
let g:UltiSnipsExpandTrigger="<c-l>"
let g:UltiSnipsJumpForwardTrigger="<c-b>"
let g:UltiSnipsJumpBackwardTrigger="<c-z>"
let g:UltiSnipsEditSplit="vertical"
```
by default, I get
`g:UltiSnipsSnippetDirectories = ['UltiSnips']`

When I edit code in one of my projects, I run `:UltiSnipsEdit` which opens up a new file in `$PWD/UltiSnips/$ft.snippets`, where `$PWD` happens to be the project root, e.g. `~/src/project_name` . I create a new UltiSnip `foo` in that file and save the .snippets, switch over to the split with my project source code, type `foo<c-l>` and nothing happens.

After digging around in the code I found that _file_to_edit resolves here: https://github.com/SirVer/ultisnips/blame/master/pythonx/UltiSnips/snippet_manager.py#L749
`_get_file_to_edit` calls `find_snippet_files(ft, 'UltiSnips')`, and `find_snippet_files appends `os.path.realpath('UltiSnips')` to the list of paths, i.e. `$PWD/UltiSnips`. So far, so good.

When I actually try to expand my snippet `find_all_snippet_files` appends 'UltiSnips' to everything in `_vim.eval('&runtimepath')`:
https://github.com/SirVer/ultisnips/blob/master/pythonx/UltiSnips/snippet/source/file/ultisnips.py#L39

but runtimepath doesn't contain `$PWD`, so it doesn't find my file. Changing the above line to be:
```
check_dirs = _vim.eval('&runtimepath').split(',') + [os.getcwd(),]
```
 fixes the issue for me, but I don't suggest fixing it that way as it would just add more confusion.

It would be much better if you refactored the code so that `_file_to_edit` searched a proper subset of the paths known to find_all_snippet_files (e.g. excluding runtimepath) and not manually construct paths itself (e.g. like this: https://github.com/SirVer/ultisnips/blob/master/pythonx/UltiSnips/snippet_manager.py#L755-L778 )

**Comment by froody** [comment]
It looks like this might be the root cause of some of the issues reported in https://github.com/SirVer/ultisnips/issues/512

**Comment by Pand9** [comment]
Thanks, I've had similar problem: https://github.com/SirVer/ultisnips/issues/919 . I worked around it by setting `let g:UltiSnipsSnippetDirectories=["$HOME/.vim/snips"]`. Note that I used `$HOME`. Surprisingly it works, and `~` doesn't. Absolute path also works.

**Comment by fvgs** [comment]
This is consistent with my own findings that the defaults for `g:UltiSnipsSnippetsDir` and `g:UltiSnipsSnippetDirectories` do not correspond in a sensible way.

Most noticeably, running `:UltiSnipsEdit` will, by default, create an UltiSnips directory and snippets file relative to the cwd. This is rarely what you want as its likely to store the snippets in your project directory. Not only that, but UltiSnips does not by default look for a snippets directory in the cwd. Thus, the new snippets created using `:UltiSnipsEdit` won't even be found.

These problems can be solved by customizing the above mentioned variables. However, neither the quick start nor the videos mention these variables. The second video actually mentions new snippets are created in `~/.vim/UltiSnips/` although this is not the default and there is no mention of configuring the appropriate variables.

At the moment, a new user must find the above mentioned variables in the docs and set them accordingly in order for UltiSnips to properly function with custom snippets. This suggests a need for sane defaults.

@froody seems to have identified the underlying cause of this behavior, but I wonder if maybe this is only an issue that has emerged in newer versions of Vim for which the default `runtimepath` may be set differently than in previous versions of Vim?

I'm running Vim **8.0.1750**

For those experiencing this issue, you can do one of the following:

```vim
let g:UltiSnipsSnippetsDir = $HOME . "/.vim/UltiSnips"
```
or
```vim
let g:UltiSnipsSnippetDirectories = [$HOME . "/.vim/UltiSnips"]
```

either will suffice, see the docs for specifics on why.

**Comment by SirVer** [comment]
Hopefully fixed by #1138

**Comment by Zackhardtoname** [comment]
I'd suggest putting the file in `~/.vim/pythonx/` which gets scanned automatically by vim. In this case, we don't have to define an environment variable.

The only thing to do in the case is to make sure only .tex files it I guess.
